### PR TITLE
powerarmor balance

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
there has been almost no use of power armor, and that is due to how awful it is.
This PR is buffing power armor with the hopes more people will use it.

## How This Contributes To The Skyrat Roleplay Experience
making this more relevant can promote forging, interactions with robotics for the suits, etc.

## Changelog

:cl:
balance: makes some of the power armor modules cost less
balance: caps power armor armor to 80
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/powerarmor/code/powerarmor.dm [hunk 1]
```diff
@@ -312,24 +312,19 @@
...
-		living_wearer.adjustToxLoss(-3)
-	if(healing_upgraded[4] && living_wearer.getOxyLoss())
-		living_wearer.adjustOxyLoss(-3)
-	if(healing_upgraded[5] && living_wearer.getStaminaLoss())
-		living_wearer.adjustStaminaLoss(-3)
+	living_wearer.adjustBruteLoss(-healing_upgraded[1])
+	living_wearer.adjustFireLoss(-healing_upgraded[2])
+	living_wearer.adjustToxLoss(-healing_upgraded[3])
+	living_wearer.adjustOxyLoss(-healing_upgraded[4])
+	living_wearer.adjustStaminaLoss(-healing_upgraded[5])
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/powerarmor/code/powerarmor.dm:327]: Those numbers should all be defines.

File: modular_skyrat/modules/powerarmor/code/powerarmor.dm [hunk 2]
```diff
@@ -384,37 +379,49 @@
...
+				if(healing_upgraded[4] >= 6)
 					return
-				healing_upgraded[4] = TRUE
+				healing_upgraded[4]++
 			if("stamina healing")
-				if(healing_upgraded[5])
+				if(healing_upgraded[5] >= 6)
 					return
-				healing_upgraded[5] = TRUE
+				healing_upgraded[5]++
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/powerarmor/code/powerarmor.dm:424]: A lot of magic numbers, I do not know what a `healing_upgraded[3]` or an `armor_upgraded[1]` is.

Also, I'm fairly confident lists start at 0 so this is just a bunch of runtimes for index out of bounds waiting to happen.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Needs some improvement when it comes to all the magic numbers, make use of more defines for the different armor types and different healing types.

File: modular_skyrat/modules/powerarmor/code/powerarmor.dm (hunk 2 above)
Comment (ChungusGamer666) [reply modular_skyrat/modules/powerarmor/code/powerarmor.dm:424]: Lists start at 1 in byond code.

File: modular_skyrat/modules/powerarmor/code/powerarmor.dm (hunk 1 above)
Comment (ChungusGamer666) [reply modular_skyrat/modules/powerarmor/code/powerarmor.dm:327]: No, they shouldn't, as they can get changed in runtime. Using a list is acceptable, but i'd recommend using an associated list (damagetype = damage healing value)

Comment (GoldenAlpharex) [reply modular_skyrat/modules/powerarmor/code/powerarmor.dm:327]: > No, they shouldn't, as they can get changed in runtime. Using a list is acceptable, but i'd recommend using an associated list (damagetype = damage healing value)

That isn't what I meant. `1` should be `BRUTE_HEALING` or something like that. That way, it's actually clearer which is which.

**Pull request closed without merging**